src/analysermodel.cpp

```diff
@@ -348,4 +377,35 @@ bool AnalyserModel::needAcothFunction() const
     return mPimpl->mNeedAcothFunction;
 }
 
+bool AnalyserModel::areEquivalentVariables(const VariablePtr &variable1,
+                                           const VariablePtr &variable2)
+{
+    // We used to have a utilities method which implementation was:
+    //
+    //     return (variable1 == variable2)
+    //            || variable1->hasEquivalentVariable(variable2, true);
+    //
+    // However, a call to Variable::hasEquivalentVariable() can be time
+    // consuming. So, here, we stripped down the implementation of that method
+    // (and of others that it calls) and cache its results for future re-use.
```

Comment:
Is this not a bad thing to do? surely `Variable::hasEquivalentVariable()` should be the only place doing this kind of check? And here is just caching the results of lookups to improve repeated calls?

Reply:
Yes, this is definitely a bad thing to do, but one that I did at some point to gain as much speed up as possible. However, my implementation back then was clearly different since I have now just tested again using the original utility and, sure, it's slower, but somewhat negligible compared to the case where we don't cache the results. So, yes, I have "reverted" things a bit in commits bb742dcd4 and 403112a6b.